node/http.go

```diff
@@ -112,6 +113,70 @@ func makeRPCOverWebsocket(host string, port uint16, servers map[string]*jsonrpc.
 	return makeHTTPService(host, port, cors.Default().Handler(mux))
 }
 
+type ipcService struct {
+	handlers []*jsonrpc.Ipc
+}
+
+var _ service.Service = (*ipcService)(nil)
+
+// Run is a method of the ipcService struct that starts the execution of IPC handlers in parallel.
+// It takes context as a parameter to control the execution and returns an error if any handler fails.
+func (ipc *ipcService) Run(ctx context.Context) error {
+	errCh := make(chan error, len(ipc.handlers))
+	defer close(errCh)
+
+	var wg conc.WaitGroup
+	defer wg.Wait()
+
+	ctx, cancel := context.WithCancelCause(ctx)
+
+	// Iterate over all registered handlers and execute them concurrently.
+	for i := range ipc.handlers {
+		handler := ipc.handlers[i]
+		wg.Go(func() {
+			cancel(handler.Run(ctx))
+		})
+	}
+
+	<-ctx.Done()
+	return context.Cause(ctx)
```

Comment:
I realized may need to do a tad more:
```go
	if err := context.Cause(ctx); err == ctx.Err() {
		return nil
	} else {
		return err
	}
```
This way we don't return spurious `context.Canceled`s and flood the logs with errors when a user types Ctrl-c.

Reply:
Why `err == ctx.Err()`  instead of `errors.Is` ?

Reply:
No particular reason. Both should work since `context.Cause` is documented to return the same as `ctx.Err()` when the cause is `nil` or when a parent was canceled.